### PR TITLE
APP-142: [C] Fix gpiocontrol slave demo

### PR DESCRIPTION
**Pull request by Dubland**
/cc @Slash32 @vadimol

### CODE REVIEW

File: common/c/targets/esp8266/target_gpio_led.c [hunk 1]
```diff
@@ -24,36 +24,55 @@
...
+
+    pGPIOConfig.GPIO_IntrType = GPIO_PIN_INTR_DISABLE;
+    pGPIOConfig.GPIO_Mode = GPIO_Mode_Output;
+    pGPIOConfig.GPIO_Pullup = GPIO_PullUp_EN;
+
+    for (unsigned int i = 0; i < NUM_GPIO_LEDS; i++) {
+        pGPIOConfig.GPIO_Pin = (BIT(gpios[i].number));
+        gpio_config(&pGPIOConfig);
+
+        GPIO_OUTPUT_SET(gpios[i].number, 0);
```
Comment (vadimol) [inline common/c/targets/esp8266/target_gpio_led.c]: 0 -> LOW

File: common/c/targets/esp8266/target_gpio_led.c [hunk 2]
```diff
@@ -24,36 +24,55 @@
...
     return NUM_GPIO_LEDS;
 }
 gpio_port_t *target_get_gpio_port( unsigned int led )
 {
-    /*It is necessary to realize the reading of the list of leds  */
-    return NULL;
+    if (led >= NUM_GPIO_LEDS) {
+        return NULL;
+    }
+	return &gpios[led];
```
Comment (rasendubi) [inline common/c/targets/esp8266/target_gpio_led.c]: code style

File: common/c/targets/posix/target_gpio_led.c [hunk 3]
```diff
@@ -58,6 +65,8 @@ unsigned int target_gpio_led_get_count(void)
 
 gpio_port_t *target_get_gpio_port(unsigned int led)
 {
-    /* It is necessary to realize the reading of the list of leds */
-    return NULL;
+	 if (led >= NUM_GPIO_LEDS) {
```
Comment (rasendubi) [inline common/c/targets/posix/target_gpio_led.c]: code style

File: common/c/targets/posix/target_gpio_led.c [hunk 4]
```diff
@@ -58,6 +65,8 @@ unsigned int target_gpio_led_get_count(void)
...
+        return NULL;
+    }
+	return &gpios[led];
```
Comment (rasendubi) [inline common/c/targets/posix/target_gpio_led.c]: code style

File: gpiocontrol/source/c/config.cmake [hunk 5]
```diff
@@ -14,14 +14,25 @@
...
 # Our application name.
 set(APP_NAME demo_client)
 
+# Disable unused features
+set (WITH_EXTENSION_CONFIGURATION OFF CACHE BOOL "")
+set (WITH_EXTENSION_NOTIFICATION OFF CACHE BOOL "")
+set (WITH_EXTENSION_LOGGING OFF CACHE BOOL "")
+set (WITH_EXTENSION_CONFIGURATION OFF CACHE BOOL "")
+set (WITH_EXTENSION_USER OFF CACHE BOOL "")
+set (WITH_ENCRYPTION OFF CACHE BOOL "")
```
Comment (rasendubi) [inline gpiocontrol/source/c/config.cmake]: code style (there should be no space between `set` and `(`)

File: gpiocontrol/source/c/config.cmake [hunk 6]
```diff
@@ -14,14 +14,25 @@
...
+
 # Target-independent flags.
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -g -Wall -Wextra")
 
+if(KAA_PLATFORM STREQUAL "esp8266")
+    set (CMAKE_BUILD_TYPE MinSizeRel)
```
Comment (rasendubi) [inline gpiocontrol/source/c/config.cmake]: code style

**Review by rasendubi — COMMENTED** [review]
Please, configure your editor to emit spaces instead of tabs.

**Review by vadimol — APPROVED** [review]

File: common/c/targets/esp8266/target_gpio_led.c [hunk 7]
```diff
@@ -63,16 +63,18 @@ bool target_gpio_led_get_state(unsigned int led)
     if (led >= NUM_GPIO_LEDS) {
         return 0;
     }
-    return (gpios[led].state != 0) ? true : false;
+    return (gpios[led].state != LOW) ? true : false;
```
Comment (rasendubi) [inline common/c/targets/esp8266/target_gpio_led.c:66]: This is equivalent to `return gpios[led].state != LOW;`

File: common/c/targets/esp8266/target_gpio_led.c [hunk 8]
```diff
@@ -24,36 +24,57 @@
...
     if (led >= NUM_GPIO_LEDS) {
         return;
     }
 
-    if (status) {
-        GPIO_OUTPUT_SET(led, HIGH);
-    } else {
-        GPIO_OUTPUT_SET(led, LOW);
-    }
+    gpios[led].state = (status) ? HIGH : LOW;
```
Comment (rasendubi) [inline common/c/targets/esp8266/target_gpio_led.c:56]: `(` and `)` are not needed.

File: common/c/targets/esp8266/target_gpio_led.c [hunk 9]
```diff
@@ -24,36 +24,57 @@
...
 }
 
 bool target_gpio_led_get_state(unsigned int led)
 {
-    /*It is necessary to realize the reading of the list of leds  */
-    return false;
+    if (led >= NUM_GPIO_LEDS) {
+        return 0;
+    }
+    return (gpios[led].state != LOW) ? true : false;
```
Comment (rasendubi) [inline common/c/targets/esp8266/target_gpio_led.c:66]: `return gpios[led].state != LOW;`

**Pull request merged**